omod/src/main/java/org/openmrs/module/queue/web/QueueEntryTransitionRestController.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.queue.web;
+
+import java.util.Date;
+import java.util.Map;
+import java.util.Optional;
+
+import org.openmrs.Concept;
+import org.openmrs.api.APIException;
+import org.openmrs.module.queue.api.QueueServicesWrapper;
+import org.openmrs.module.queue.model.Queue;
+import org.openmrs.module.queue.model.QueueEntry;
+import org.openmrs.module.queue.model.QueueEntryTransition;
+import org.openmrs.module.webservices.rest.web.ConversionUtil;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+/**
+ * The main controller that exposes additional end points for order entry
+ */
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/queue-entry-transition")
+public class QueueEntryTransitionRestController extends BaseRestController {
+	
+	public static final String QUEUE_ENTRY_TO_TRANSITION = "queueEntryToTransition";
+	
+	public static final String TRANSITION_DATE = "transitionDate";
+	
+	public static final String NEW_QUEUE = "newQueue";
+	
+	public static final String NEW_STATUS = "newStatus";
+	
+	public static final String NEW_PRIORITY = "newPriority";
+	
+	private final QueueServicesWrapper services;
+	
+	@Autowired
+	public QueueEntryTransitionRestController(QueueServicesWrapper services) {
+		this.services = services;
+	}
+	
+	@RequestMapping(method = { RequestMethod.PUT, RequestMethod.POST })
+	@ResponseStatus(HttpStatus.OK)
```

Comment:
I copied the skeleton of this implementation from the `PasswordResetController2_2` class in the webservices.rest module, looking for a pattern to follow.  Interested if we think this should return a response body or not (eg. containing the newly created queue entry, etc)

Reply:
I think the response here (and in the PasswordResetController) should be `HttpStatus.NO_CONTENT` if there's no body.

That said, for things like this where we're effectively creating a whole new entry, I do think it's better if the response returns that entry... That gives us a lot more flexibility on the frontend and allows us to potentially avoid an unnecessary GET request.

Reply:
Yeah, my gut would be to return the new entry.

Reply:
Done.  I went with the REF representation, which seems sufficient.